hpx/lcos_fwd.hpp

```diff
@@ -85,7 +87,11 @@ namespace hpx
 
     using lcos::future;
     using lcos::shared_future;
+#if defined(HPX_HAVE_DISTRIBUTED_RUNTIME)
     using lcos::promise;
+#else
+    using lcos::local::promise;
+#endif
```

Comment:
This is very dangerous as it (silently) changes semantics of user code depending on whether `HPX_HAVE_DISTRIBUTED_RUNTIME` is defined or not.

Reply:
Good point, will change.